Realm/RLMConstants.h

```diff
@@ -92,8 +94,12 @@ typedef RLM_CLOSED_ENUM(int32_t, RLMPropertyType) {
     /** Realm linking objects. See [Realm Models](https://www.mongodb.com/docs/realm/sdk/swift/fundamentals/relationships/#inverse-relationship) for more information. */
     RLMPropertyTypeLinkingObjects = 8,
 
-    RLMPropertyTypeObjectId = 10,
-    RLMPropertyTypeDecimal128 = 11
+#pragma mark - Collection types
+
+    /** Dictionary: `RLMDictionary`, `Map` (Swift) */
+    RLMPropertyTypeDictionary = 512,
```

Comment:
These aren't property types, and adding these to this enum is a significant violation of the design. This project is making the set of types which can be stored in mixed no longer the same as the primitive property types, so it needs to be introducing a new enum for that.